.travis.yml

```diff
@@ -46,6 +46,19 @@ matrix:
         - DESC="findbugs and pmd"
         - CMD="mvn clean compile pmd:check findbugs:check"
 
+    # checkstyle injection compile no error (oraclejdk8)
+    - jdk: oraclejdk8
+      env:
+        - DESC="checkstyle injection compile no error"
+        - CMD1="git clone https://github.com/checkstyle/checkstyle &&"
+        - CMD2="cp src/main/resources/com/github/checkstyle/regression/extract/ExtractInfoGeneratorTest.java "
+        - CMD3="checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/ExtractInfoGeneratorTest.java &&"
+        - CMD4="cp src/main/resources/com/github/checkstyle/regression/extract/Json.java "
+        - CMD5="checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/Json.java &&"
+        - CMD6="cd checkstyle &&"
+        - CMD7="mvn clean compile"
+        - CMD="$CMD1$CMD2$CMD3$CMD4$CMD5$CMD6$CMD7"
```

Comment:
What level of the test should be? The ones I could think out now:
1. check it is a valid json
2. check whether it is in correct format(a list, each item contains "name", "packageName" etc)
Do you have anything to add?

And, to test a json format is much hard to do with only shell script, if we won't do it in UT, then could we use other script lang like Groovy?